compiler-plugin/src/main/java/io/ballerina/stdlib/websocket/plugin/WebSocketServiceValidator.java

```diff
@@ -105,6 +160,41 @@ public void validate() {
                 !functionSet.containsKey(PluginConstants.ON_BINARY_MESSAGE)) {
             reportDiagnostic(classDefNode, PluginConstants.CompilationErrors.ON_MESSAGE_GENERATION_HINT);
         }
+        validateDispatcherMappingAnnotations(classDefNode, functionSet);
+    }
+
+    private void validateDispatcherMappingAnnotations(ClassDefinitionNode classDefNode,
+                                                      Map<String, Boolean> functionSet) {
+        Set<String> seenAnnotationValues = new HashSet<>();
+        for (Node node : classDefNode.members()) {
+            if (!node.kind().equals(SyntaxKind.OBJECT_METHOD_DEFINITION)) {
+                continue;
+            }
+            FunctionDefinitionNode funcDefinitionNode = (FunctionDefinitionNode) node;
+            Optional<String> funcName = ctx.semanticModel().symbol(funcDefinitionNode).flatMap(Symbol::getName);
+            Optional<String> annoDispatchingValue = getDispatcherMappingAnnotatedFunctionName(funcDefinitionNode, ctx);
+            if (funcName.isEmpty() || annoDispatchingValue.isEmpty()) {
+                continue;
+            }
+            if (seenAnnotationValues.contains(annoDispatchingValue.get())) {
+                Utils.reportDiagnostics(ctx, DUPLICATED_DISPATCHER_MAPPING_VALUE,
+                        funcDefinitionNode.location(), annoDispatchingValue.get());
+                continue;
+            }
+            seenAnnotationValues.add(annoDispatchingValue.get());
+            String customRemoteFunctionName = createCustomRemoteFunction(annoDispatchingValue.get());
+            if (specialRemoteMethods.contains(funcName.get())) {
```

Comment:
Use `this` keyword when referring to instance variables. Check similar places.

```suggestion
            if (this.specialRemoteMethods.contains(funcName.get())) {
```

Reply:
Fixed with [e8a82d26480ee1dbe28dee39a4c0a61edead19d2](https://github.com/ballerina-platform/module-ballerina-websocket/pull/1553/commits/e8a82d26480ee1dbe28dee39a4c0a61edead19d2)